File: app/src/test/java/org/wikipedia/test/ImmediateExecutorService.java
package org.wikipedia.test;

import androidx.annotation.NonNull;

import java.util.List;
import java.util.concurrent.AbstractExecutorService;
import java.util.concurrent.TimeUnit;

public final class ImmediateExecutorService extends AbstractExecutorService {
    @Override public void shutdown() {
        throw new UnsupportedOperationException();
    }

    @NonNull @Override public List<Runnable> shutdownNow() {
        throw new UnsupportedOperationException();
    }

    @Override public boolean isShutdown() {
        throw new UnsupportedOperationException();
    }

    @Override public boolean isTerminated() {
        throw new UnsupportedOperationException();
    }

    @Override public boolean awaitTermination(long l, @NonNull TimeUnit timeUnit) {
        throw new UnsupportedOperationException();
    }

    @Override public void execute(@NonNull Runnable runnable) {
        runnable.run();
    }
}

File: app/src/test/java/org/wikipedia/test/ImmediateExecutorService.kt
package org.wikipedia.test

import java.lang.UnsupportedOperationException
import java.util.concurrent.AbstractExecutorService
import java.util.concurrent.TimeUnit

class ImmediateExecutorService : AbstractExecutorService() {
    override fun shutdown() {
        throw UnsupportedOperationException()
    }

    override fun shutdownNow(): MutableList<Runnable?> {
        throw UnsupportedOperationException()
    }

    override fun isShutdown(): Boolean {
        throw UnsupportedOperationException()
    }

    override fun isTerminated(): Boolean {
        throw UnsupportedOperationException()
    }

    override fun awaitTermination(l: Long, timeUnit: TimeUnit): Boolean {
        throw UnsupportedOperationException()
    }

    override fun execute(runnable: Runnable) {
        runnable.run()
    }
}
